packages/atomic/cypress/integration/common-actions.ts

```diff
@@ -0,0 +1,53 @@
+function observeRecursively(
+  element: Node,
+  callback: (mutations: MutationRecord[]) => void,
+  isAlreadyObserved = false
+): () => void {
```

Comment:
I'm honestly not sure how to simplify this. The problem is that `MutationObserver` doesn't observe the contents of shadowDOMs, so I made this function to recursively observe:
* Changes in the root element (on which this was first called)
* Changes in every shadowRoot (recursively)